contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -291,6 +291,28 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
         Ok(())
     }
 
+    #[when_not_paused]
+    fn register_custom_token(
+        env: &Env,
+        salt: BytesN<32>,
+        token_address: Address,
+        token_manager_type: TokenManagerType,
+    ) -> Result<BytesN<32>, ContractError> {
+        // Custom token managers can't be deployed with native interchain token type, which is reserved for interchain tokens
+        ensure!(
+            token_manager_type != TokenManagerType::NativeInterchainToken,
+            ContractError::InvalidTokenManagerType
+        );
+        let token_id = Self::interchain_token_id(env, Address::zero(env), salt);
```

Comment:
Incorrect derivation. See how custom token IDs are derived in the ITS factory. The deployer/caller needs to be passed in, since this is not a canonical deployment

Reply:
Note how msg.sender is used in EVM ITS to identify mismatching implementations